packages/next/src/build/webpack/loaders/next-app-loader.ts

```diff
@@ -553,13 +554,8 @@ const nextAppLoader: AppLoader = async function nextAppLoader() {
       return existingFiles.has(fileName)
     }
     try {
-      const files = await fs.readdir(dirname, { withFileTypes: true })
-      const fileNames = new Set<string>()
-      for (const file of files) {
-        if (file.isFile()) {
-          fileNames.add(file.name)
-        }
-      }
```

Comment:
When writing tests I noticed that this area suffered form the same issue with symbolic links. It wouldn't recognise that `layout.tsx` existed. 

This fuelled my decision to re-use the flat-readdir function rather than fix this code here too.